gordo_components/dataset/datasets.py

```diff
@@ -157,13 +165,23 @@ def get_data(self) -> Tuple[pd.DataFrame, Optional[pd.DataFrame]]:
 
         # Resample if we have a resolution set, otherwise simply join the series.
         if self.resolution:
-            data = self.join_timeseries(
-                series_iter,
-                self.from_ts,
-                self.to_ts,
-                self.resolution,
-                aggregation_methods=self.aggregation_methods,
-            )
+            try:
+                data = self.join_timeseries(
+                    series_iter,
+                    self.from_ts,
+                    self.to_ts,
+                    self.resolution,
+                    aggregation_methods=self.aggregation_methods,
+                )
+                if len(data) <= self.row_threshold:
+                    raise InsufficientDataError(
+                        f"The length of the generated DataFrame ({len(data)}) does not exceed the "
+                        f"specified required threshold for number of rows ({self.row_threshold})."
+                    )
```

Comment:
I was thinking rather this should be outside of here and after the possible joining without resolution. That way both are checked.

Reply:
OK, got it. See what you think now :+1: